### ISSUE TITLE
Avoid duplicate code in constructors by using the base class constructors

### ISSUE THREAD

**Issue by KonstantinSchubert** [gammacombo/gammacombo#46]
We should also think about using initialization lists instead of initializing variables in the constructors:
http://stackoverflow.com/questions/926752/why-should-i-prefer-to-use-member-initialization-list

Some notes to myself because it has been a while since I last used C++:

Some info about calling base class constructors:
http://stackoverflow.com/questions/120876/c-superclass-constructor-calling-rules

Basically, we can either implement the default constructor of the base class and initialize most of the stuff there. It should then be called automatically in the constructor of the subclass, if I got it right.
Or we can call the base class constructor with arguments explicitely in the subclass' initializer list.

**Comment by matthewkenzie** [comment]
Should be resolved by pull request #54
